src/test/java/org/codeforamerica/messaging/controllers/TemplateControllerTest.java

```diff
@@ -0,0 +1,139 @@
+
+@WebMvcTest(TemplateController.class)
+@Import(SecurityConfiguration.class)
+public class TemplateControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private TemplateService templateService;
+
+    private final String TEMPLATE_NAME_WITHOUT_VARIANTS = "Template name without variants";
+    private final String TEMPLATE_NAME_WITH_VARIANTS = "Template name with variants";
+    private final Template TEMPLATE_WITHOUT_VARIANTS = TestData.aTemplate()
+            .name(TEMPLATE_NAME_WITHOUT_VARIANTS)
+            .build();
+    private final Template TEMPLATE_WITH_VARIANTS = TestData.aTemplate()
+            .name(TEMPLATE_NAME_WITH_VARIANTS)
+            .build();
+
+    @BeforeEach
+    void setup() {
+        TestData.addVariantsToTemplate(TEMPLATE_WITH_VARIANTS);
+    }
+
+    @Test
+    public void whenUnauthenticated_thenUnauthorized() throws Exception {
+        mockMvc.perform(get("/api/v1/templates"))
+                .andExpect(MockMvcResultMatchers.status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void whenAuthenticatedAndNoTemplates_thenNotFound() throws Exception {
+        Mockito.when(templateService.getTemplateList())
+                .thenReturn(Collections.emptyList());
+
+        mockMvc.perform(get("/api/v1/templates")
+                        .with(httpBasic("user", "password")))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    @Test
+    @WithMockUser
+    public void whenAuthenticatedAndNoTemplatesWithMatchingName_thenNotFound() throws Exception {
+        Mockito.when(templateService.getTemplateByName("something"))
+                .thenReturn(Optional.empty());
+
+        mockMvc.perform(get("/api/v1/templates?name=something")
+                        .with(httpBasic("user", "password")))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    @Test
+    @WithMockUser
+    public void whenAuthenticatedAndNoTemplatesWithMatchingId_thenNotFound() throws Exception {
+        Mockito.when(templateService.getTemplateById(1L))
+                .thenReturn(Optional.empty());
+
+        mockMvc.perform(get("/api/v1/templates/1")
+                        .with(httpBasic("user", "password")))
+                .andExpect(MockMvcResultMatchers.status().isNotFound());
+    }
+
+    @Test
+    @WithMockUser
+    public void whenAuthenticatedAndTemplatesExist_thenReturnTemplatesAndVariants() throws Exception {
+        Mockito.when(templateService.getTemplateList())
+                .thenReturn(List.of(TEMPLATE_WITH_VARIANTS, TEMPLATE_WITHOUT_VARIANTS));
+
+        mockMvc.perform(get("/api/v1/templates")
+                        .with(httpBasic("user", "password")))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(content().string(containsString(TEMPLATE_NAME_WITHOUT_VARIANTS)))
```

Comment:
I was thinking this would be more flexible if we wanted to ever change the response structure. As long as the template name, body, and subject and in the response then it doesn't matter what format it is in